### PR TITLE
[ID-51] load and display location address

### PR DESCRIPTION
**Pull request by lqchuong**

### CODE REVIEW

File: packages/nartus_geocoder/build/unit_test_assets/AssetManifest.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (suesitran) [inline packages/nartus_geocoder/build/unit_test_assets/AssetManifest.json]: what is this??

File: packages/nartus_geocoder/build/unit_test_assets/FontManifest.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (suesitran) [inline packages/nartus_geocoder/build/unit_test_assets/FontManifest.json]: what is this??

File: lib/features/home/bloc/location_bloc.dart [hunk 3]
```diff
@@ -12,18 +14,24 @@ const String _dateFormat = 'dd-MMM-yyyy';
...
 
   Future<void> requestCurrentLocation() async {
+    final String dateDisplay = DateFormat(_dateFormat).format(DateTime.now());
     try {
       final LocationDetails data = await _locationService.getCurrentLocation();
-      final String dateDisplay = DateFormat(_dateFormat).format(DateTime.now());
-
-      emit(LocationReadyState(
-          LatLng(data.latitude, data.longitude), dateDisplay));
+      await _geocoderService
```
Comment (suesitran) [inline lib/features/home/bloc/location_bloc.dart]: `await` and `then` at the same time? not wrong, but not a good practice

File: lib/features/home/bloc/location_bloc.dart [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import 'package:equatable/equatable.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:geocoding/geocoding.dart';
```
Comment (suesitran) [inline lib/features/home/bloc/location_bloc.dart]: u already have nartus_geocoder as a service, why do u still need to use this package here?

File: lib/features/home/bloc/location_bloc.dart [hunk 5]
```diff
@@ -49,8 +57,18 @@ class LocationBloc extends Cubit<LocationState> {
 
   Future<void> requestDefaultLocation() async {
     final String dateDisplay = DateFormat(_dateFormat).format(DateTime.now());
-
-    emit(LocationReadyState(_defaultLocation, dateDisplay));
+    try {
```
Comment (suesitran) [inline lib/features/home/bloc/location_bloc.dart:73]: same as above

File: lib/features/home/content_panel/contents_bottom_panel_view.dart [hunk 6]
```diff
@@ -164,4 +168,37 @@ class _ContentsBottomPanelViewState extends State<ContentsBottomPanelView>
...
+              '${infoLocation.street}, ${infoLocation.subLocality}, ${infoLocation.subAdministrativeArea}, ${infoLocation.administrativeArea}, ${infoLocation.country} ',
+          latitude: widget._location.latitude,
+          longitude: widget._location.longitude,
+          borderRadius: BorderRadius.circular(12),
+        );
+      } else {
+        return LocationView(
+          locationIconSvg: Assets.images.idLocationIcon,
+          businessName: widget._infoLocation?.name,
+          address:
```
Comment (suesitran) [inline lib/features/home/content_panel/contents_bottom_panel_view.dart]: please compute the `business name` and `location address` in `nartus_geocoder` package, and use that string here.

File: lib/features/home/home_screen.dart [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:geocoding/geocoding.dart';
```
Comment (suesitran) [inline lib/features/home/home_screen.dart]: unused import?

File: lib/features/home/widgets/google_map.dart [hunk 8]
```diff
@@ -227,7 +227,8 @@ class _GoogleMapViewState extends State<GoogleMapView>
       if (!_streamController.isClosed) {
         markers.add(Marker(
             markerId: const MarkerId(baseMarkerCurrentLocationId),
-            position: widget.currentLocation,
+            position: LatLng(widget.currentLocation.latitude,
```
Comment (suesitran) [inline lib/features/home/widgets/google_map.dart:234]: why is this change needed?

File: packages/nartus_geocoder/lib/src/geocoder_service.dart [hunk 9]
```diff
@@ -0,0 +1,13 @@
+part of "../nartus_geocoder.dart";
+
+class GeocoderService {
+  Future<Placemark?> getCurrentPlaceCoding(
+      double latitude, double longitude) async {
+    List<Placemark> placemarks = await GeocodingPlatform.instance
```
Comment (suesitran) [inline packages/nartus_geocoder/lib/src/geocoder_service.dart]: instead of returning a `List<Placemark>`, you can return an object which has `String? businessName`, and `String? address`

File: packages/nartus_geocoder/pubspec.yaml [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+homepage:
+
+environment:
+  sdk: ">=2.19.1 <3.0.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  geocoding: ^2.0.5
+dev_dependencies:
```
Comment (suesitran) [inline packages/nartus_geocoder/pubspec.yaml:13]: add an empty line between `dependencies` and `dev_dependencies` group. It just look nicer

File: packages/nartus_geocoder/test/nartus_geocoder_test.dart [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+        subLocality: 'Ho Chi Minh City',
+        thoroughfare: 'Phan Chu Trinh',
+      );
+
+      when(
+        mockGeocoder.getCurrentPlaceCoding(10.7725, 106.6980),
+      ).thenAnswer((_) => Future.value(placemarkDefault));
+      final futurePlacemart =
+          await mockGeocoder.getCurrentPlaceCoding(10.7725, 106.6980);
+      expect(futurePlacemart, placemarkDefault);
```
Comment (suesitran) [inline packages/nartus_geocoder/test/nartus_geocoder_test.dart]: ur code check for condition on the length of the list, not the detail of the list

File: test/features/home/bloc/location_bloc_test.dart [hunk 12]
```diff
@@ -10,7 +10,7 @@ import 'location_bloc_test.mocks.dart';
 @GenerateMocks(<Type>[LocationService])
 void main() {
   final MockLocationService service = MockLocationService();
-
+  TestWidgetsFlutterBinding.ensureInitialized();
```
Comment (suesitran) [inline test/features/home/bloc/location_bloc_test.dart:16]: this should be the first line in `main()`

File: lib/features/home/content_panel/contents_bottom_panel_view.dart [hunk 13]
```diff
@@ -18,10 +21,18 @@ class ContentsBottomPanelController extends ChangeNotifier {
...
 class ContentsBottomPanelView extends StatefulWidget {
+  final Placemark? _infoLocation;
+  final LatLng _location;
   final ContentsBottomPanelController controller;
 
-  const ContentsBottomPanelView({required this.controller, Key? key})
-      : super(key: key);
+  const ContentsBottomPanelView(
+      {required this.controller,
+      required location,
```
Comment (suesitran) [inline lib/features/home/content_panel/contents_bottom_panel_view.dart]: always specify the type

**Review by suesitran — CHANGES_REQUESTED** [review]

File: lib/features/home/bloc/location_bloc.dart (hunk 3 above)
Comment (lqchuong) [reply lib/features/home/bloc/location_bloc.dart]: i will choose one of them

File: packages/nartus_geocoder/build/unit_test_assets/AssetManifest.json (hunk 1 above)
Comment (lqchuong) [reply packages/nartus_geocoder/build/unit_test_assets/AssetManifest.json]: i will recheck

File: packages/nartus_geocoder/lib/src/geocoder_service.dart (hunk 9 above)
Comment (lqchuong) [reply packages/nartus_geocoder/lib/src/geocoder_service.dart]: I got it

**Review by suesitran — APPROVED** [review]
